dbt/utils.py

```diff
@@ -79,9 +79,18 @@ def compiler_warning(model, msg):
 
 
 def model_immediate_name(model, non_destructive):
-    "The name of the model table/view within the transaction"
+    """
+    Returns the name of the model relation within the transaction. This is
+    useful for referencing the model in pre or post hooks. Non-destructive
+    models aren't created with temp suffixes, nor are incremental models or
+    seeds.
+    """
```

Comment:
general comment: i implemented another workaround in Relation.render (https://github.com/fishtown-analytics/dbt/blob/development/dbt/adapters/default/relation.py#L138) to skip using the immediate name. materialization code should create relations with the correct identifiers -- if someone implements incremental differently, e.g. create, then atomic swap, then drop, using the immediate name could break it. our long term goal should be to get rid of this fn

Reply:
yeah, looking forward to burning this code path down. Hopefully as a part of https://github.com/fishtown-analytics/dbt/issues/653?